Unsplash/Unsplash/Coordinator/HomeCoordinator.swift

```diff
@@ -0,0 +1,20 @@
+//
+//  HomeCoordinator.swift
+//  Unsplash
+//
+//  Created by Nha Pham on 8/25/20.
+//  Copyright © 2020 Nha Pham. All rights reserved.
+//
+
+import UIKit
+struct HomeCoordinator: Coordinator {
+    var navigationController: UINavigationController
+
+    func start() {
+        guard let homeViewController = HomeViewController.instantiate(Storyboard.Name.Home) as? HomeViewController else {
+            return
+        }
+        homeViewController.coordinator = self
```

Comment:
Em xem lại đoạn này nhé, đang chưa đúng mô hình lắm. Coi ViewController như View thì View sẽ chứa coordinator và coordinator hoàn toàn không biết view nào chứa nó cả

Reply:
thế làm sao xét được coordinator của viewcontroller đó được a?

Reply:
e tưởng thằng viewcontroller cần phải có biến coordinator thì nó mới gọi được hàm của coordinator để chuyển màn hình được chứ

Reply:
Ở trong ViewController em khởi tạo Coordinator và sử dụng bên trong nó. Như vậy thì bên trong coordinator sẽ không biết ViewController nào gọi nó cả

Reply:
à ok a thế e sửa lại ạ

Reply:
nhưng mà như vậy có hơi rắc rối không ạ tại e xem nhiều clip hướng thì người ta vẫn set như này

Reply:
https://www.hackingwithswift.com/articles/71/how-to-use-the-coordinator-pattern-in-ios-apps

Reply:
Đoạn này anh thấy em đang hiểu sai rồi. Ở trong ví dụ em gửi là 3 viewcontrollers nhỏ hơn đang có 1 biến MainCoordinator nên mới gán `vc.coordinator = self`.

Ví dụ ở màn Home em chuyển sang màn Detail thì trong hàm `toDetail` của HomeCoordinator mới khởi tạo DetailViewController và set DetailCoordinator cho DetailViewController được tạo ra
```swift
func toDetail(id: Int) {
    let vc = DetailViewController.instantiate()
    vc.coordinator = DetailCoordinator()
    navigationController.pushViewController(vc, animated: true)
}
```

Reply:
final class HomeViewController: UIViewController {
    weak var coordinator: HomeCoordinator?
    var coordinator: HomeCoordinator?

    override func viewDidLoad() {
        super.viewDidLoad()

Reply:
là sao a e chưa hiểu lắm trong homeViewController của e cũng có biến coordinator nên e mới gán như vậy hay là mình phải khởi tạo biến mới à a?

Reply:
![image](https://user-images.githubusercontent.com/46532582/91413175-eee42180-e874-11ea-86c7-9be71d317197.png)

Reply:
![image](https://user-images.githubusercontent.com/46532582/91413295-19ce7580-e875-11ea-8f67-5e35d50ef121.png)
e phải làm như thế này á a

Reply:
![image](https://user-images.githubusercontent.com/46532582/91414755-32d82600-e877-11ea-9705-b7ce61cc6e1c.png)
ý a là khởi tạo biến như thế này à a?